qutip/bloch.py

```diff
@@ -394,6 +401,128 @@ def add_annotation(self, state_or_vector, text, **kwargs):
                                  'text': text,
                                  'opts': kwargs})
 
+    def add_arc(self, start, end, fmt="b", steps=None, **kwargs):
+        """Adds an arc between two points on a sphere. The arc is set to be
+        blue solid curve by default.
+
+        The start and end points must be on the same sphere (i.e. have the
+        same radius) but need not be on the unit sphere.
+
+        Parameters
+        ----------
+        start : Qobj or array-like
+            Array with cartesian coordinates of the first point, or a state
+            vector or density matrix that can be mapped to a point on or
+            within the Bloch sphere.
+        end : Qobj or array-like
+            Array with cartesian coordinates of the second point, or a state
+            vector or density matrix that can be mapped to a point on or
+            within the Bloch sphere.
+        fmt : str, default: "b"
+            A matplotlib format string for rendering the arc.
+        steps : int, default: None
+            The number of segments to use when rendering the arc. The default
+            uses one segment per 1/4 degree of the arc.
```

Comment:
I believe this is not strictly correct and not obvious how to program with the algorithm implemented. The reason is that we parametrize the _line_ that goes from start to end with equidistant points. When these are normalized to be in the arc they will not be separated by the same degree. I would instead just way something on the lines:
"By default we chose a variable number of points that depends on the length of the arc." and not go into the details of how they are drawn.

Reply:
To quote the Zen of Python "If the implementation is hard to explain, it's a bad idea." so I changed the implementation to make it simpler to explain. :)